test/e2e/upgrades/apps/deployments.go

```diff
@@ -84,9 +85,6 @@ func (t *DeploymentUpgradeTest) Setup(f *framework.Framework) {
 
 	rs, err := deploymentutil.GetNewReplicaSet(deployment, c.ExtensionsV1beta1())
 	framework.ExpectNoError(err)
-	if rs == nil {
```

Comment:
Checking whether `rs == nil` is redundant. `WaitForDeploymentStatusValid` or `WaitForDeploymentStatus` has already done so.